mypy/types.py

```diff
@@ -3390,3 +3392,29 @@ def callable_with_ellipsis(any_type: AnyType, ret_type: Type, fallback: Instance
         fallback=fallback,
         is_ellipsis_args=True,
     )
+
+
+def store_argument_type(
+    defn: FuncItem, i: int, typ: CallableType, named_type: Callable[[str, list[Type]], Instance]
+) -> None:
+    arg_type = typ.arg_types[i]
+    if typ.arg_kinds[i] == ARG_STAR:
+        if isinstance(arg_type, ParamSpecType):
```

Comment:
(Not directly related to this PR.) I wonder if it would make sense to add a TODO about also handling ParamSpecType? Or is there nothing we need do here (sorry, don't remember the details)?

Reply:
It already works in simple cases, but also there is a TODO in `expand_typevars()` (for type variables with values). Just in case I will add a TODO here as well.

Reply:
Actually, it looks like it works for type vars with values also, so probably we are fine here.